### PR TITLE
Make mouse hover over take chromatic abberation into account

### LINKED ISSUE #1394: Make cursor hover over take the barrel distortion into account
Added in this PR: https://github.com/Revolutionary-Games/Thrive/pull/1385

### PR DESCRIPTION
**Pull request by dligr**
**Brief Description of What This PR Does**

Makes mouse hover over menu take screen distortion (from chromatic abberation) into account, so that the menu shows what visually is under cursor.

**Related Issues**

Closes #1394

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by AverageNerdz** [comment]
This one seems to work much better i tested it for a while and it didn't impact the performances as much as the last one did

**Review by AverageNerdz — APPROVED** [review]
This one works much better, i tested it for a while and i can state that it doesn't affect performances as much as the previous one did.

File: src/general/utils/ScreenUtils.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿using Godot;
+
+/// <summary>
+///   Utility for screen and view-related effects
+/// </summary>
+public static class ScreenUtils
+{
+    /// <summary>
+    ///   Applies barrel distortion to the given position
```
Comment (hhyyrylainen) [inline src/general/utils/ScreenUtils.cs:9]: I'd like to see a remarks section added here saying that this must match the math in `Chromatic.gdshader`.

Also I'd like to see a comment in `Chromatic.gdshader` saying that if the math is updated then the math in this file needs to be updated.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Thanks for contributing to Thrive. This already works very well, but I noticed when testing that if I crank the chromatic aberration slider to the max, that affects the distortion and seems to cause this math to fail. So maybe this needs to additionally take the chromatic effect strength from settings into account? Also just a tiny nitpick on the code side that placing warning comments on the related pieces of code that must match would be nice.

File: src/general/utils/ScreenUtils.cs (hunk 1 above)
Comment (dligr) [reply src/general/utils/ScreenUtils.cs:9]: Added the comments now

**Comment by dligr** [comment]
I investigated the problem with maths in BarrelDistortion() and found out that I didn't apply one of the operations that were done in the shader script. Now (according to my testing) the distortion to cursor works almost perfect on any setting of chromatic abberation.

File: third_party/JoltPhysics
Comment (hhyyrylainen) [inline third_party/JoltPhysics:1]: I think you accidentally downgraded this git submodule.

**Comment by hhyyrylainen** [comment]
How does this work with all different distortion values when the CPU-side distortion calculation isn't given `Settings.Instance.ChromaticAmount`? Because as far as I understand that also impacts the distortion amount calculated in the shader, so a function that doesn't take that into account cannot be fully accurate if comparing the chromatic amount slider being at minimum vs when at maximum.

**Comment by dligr** [comment]
`BarrelDistortion()` is given `Settings.Instance.ChromaticAmount` through `ApplyScreenEffects()` (as `distortion` argument) in both `MicrobeCamera` and `MicrobeInspectInfo`, so this setting is taken into account.

**Comment by hhyyrylainen** [comment]
Ah you are correct. I must have already forgotten that code existed as in testing I noticed the result not being correct at max distortion setting. I shall re-test this. If the issue is solved the only problem is the submodule downgrade.

**Review by hhyyrylainen — APPROVED** [review]
This seems to now fully work. The actual math is way too complex for me to analyze but the code is neatly split into methods. 

Thanks for the long-awaited fix for this problem.

**Pull request merged**